base/dsfs/load.go

```diff
@@ -0,0 +1,302 @@
+	// if we don't have the dataset locally, IPFS will reach out onto the d.web to
+	// attempt to resolve previous hashes. capping the duration yeilds quicker results.
+	// TODO (b5) - The proper way to solve this is to feed a local-only IPFS store
+	// to this entire function, or have a mechanism for specifying that a fetch
+	// must be local
+	ctx, cancel := context.WithTimeout(ctx, OpenFileTimeoutDuration)
+	defer cancel()
+
+	ds, err := LoadDatasetRefs(ctx, store, path)
+	if err != nil {
+		log.Debugf("loading dataset: %s", err)
+		return nil, fmt.Errorf("loading dataset: %w", err)
+	}
+	if err := DerefDataset(ctx, store, ds); err != nil {
+		log.Debug(err.Error())
+		return nil, err
+	}
+
+	return ds, nil
+}
+
+// LoadDatasetRefs reads a dataset from a content addressed filesystem without
+// dereferencing components
+func LoadDatasetRefs(ctx context.Context, fs qfs.Filesystem, path string) (*dataset.Dataset, error) {
+	pathWithBasename := PackageFilepath(fs, path, PackageFileDataset)
+	log.Debugw("LoadDatasetPath", "packageFilepath", pathWithBasename)
+	data, err := fileBytes(fs.Get(ctx, pathWithBasename))
+	if err != nil {
+		log.Debug(err.Error())
+		return nil, fmt.Errorf("reading %s file: %w", PackageFileDataset.String(), err)
+	}
+
+	ds, err := dataset.UnmarshalDataset(data)
+	if err != nil {
+		log.Debug(err.Error())
+		return nil, fmt.Errorf("unmarshaling %s file: %w", PackageFileDataset.String(), err)
+	}
+
+	// assign path to retain reference to the path this dataset was read from
+	ds.Path = path
+
+	return ds, nil
+}
+
+// DerefDataset attempts to fully dereference a dataset
+func DerefDataset(ctx context.Context, store qfs.Filesystem, ds *dataset.Dataset) error {
+	if err := DerefMeta(ctx, store, ds); err != nil {
+		return err
+	}
+	if err := DerefStructure(ctx, store, ds); err != nil {
+		return err
+	}
+	if err := DerefTransform(ctx, store, ds); err != nil {
+		return err
+	}
+	if err := DerefViz(ctx, store, ds); err != nil {
+		return err
+	}
+	if err := DerefReadme(ctx, store, ds); err != nil {
+		return err
+	}
+	if err := DerefStats(ctx, store, ds); err != nil {
+		return err
+	}
+	return DerefCommit(ctx, store, ds)
+}
+
+// LoadBody loads the data this dataset points to from the store
+func LoadBody(ctx context.Context, fs qfs.Filesystem, ds *dataset.Dataset) (qfs.File, error) {
+	return fs.Get(ctx, ds.BodyPath)
+}
+
+// DerefCommit derferences a dataset's Commit element if required
+// should be a no-op if ds.Structure is nil or isn't a reference
+func DerefCommit(ctx context.Context, store qfs.Filesystem, ds *dataset.Dataset) error {
+	if ds.Commit != nil && ds.Commit.IsEmpty() && ds.Commit.Path != "" {
+		cm, err := loadCommit(ctx, store, ds.Commit.Path)
+		if err != nil {
+			log.Debug(err.Error())
+			return fmt.Errorf("loading dataset commit: %w", err)
+		}
+		cm.Path = ds.Commit.Path
+		ds.Commit = cm
+	}
+	return nil
+}
+
+// loadCommit assumes the provided path is valid
```

Comment:
Comment here should reflect what the function does from the perspective of a caller. How about "loadCommit loads and unmarshals the commit data at the given path"

Same for loadMeta, loadReadme, loadStats, loadTransform, loadStructure, loadViz

Reply:
I've elected to remove the comment entirely